deployment/src/main/java/com/datastax/oss/quarkus/deployment/MetricsBuildStep.java

```diff
@@ -0,0 +1,161 @@
+
+  @BuildStep
+  void registerMetrics(
+      CassandraClientBuildTimeConfig buildTimeConfig,
+      BuildProducer<MetricBuildItem> metrics,
+      Capabilities capabilities) {
+    Map<SessionMetric, Metadata> supportedSessionMetrics = new HashedMap();
+
+    supportedSessionMetrics.put(
+        BYTES_SENT,
+        Metadata.builder()
+            .withName(BYTES_SENT.getPath())
+            .withDescription("The number and rate of bytes sent for the entire session.")
+            .withType(MetricType.METERED)
+            .build());
+
+    supportedSessionMetrics.put(
+        BYTES_RECEIVED,
+        Metadata.builder()
+            .withName(BYTES_RECEIVED.getPath())
+            .withDescription("The number and rate of bytes received for the entire session.")
+            .withType(MetricType.METERED)
+            .build());
+
+    supportedSessionMetrics.put(
+        CONNECTED_NODES,
+        Metadata.builder()
+            .withName(CONNECTED_NODES.getPath())
+            .withDescription(
+                "The number of nodes to which the driver has at least one active connection")
+            .withType(MetricType.GAUGE)
+            .build());
+
+    supportedSessionMetrics.put(
+        CQL_REQUESTS,
+        Metadata.builder()
+            .withName(CQL_REQUESTS.getPath())
+            .withDescription("The throughput and latency percentiles of CQL requests.")
+            .withType(MetricType.METERED)
+            .build());
+
+    supportedSessionMetrics.put(
+        THROTTLING_DELAY,
+        Metadata.builder()
+            .withName(THROTTLING_DELAY.getPath())
+            .withDescription("How long requests are being throttled.")
+            .withType(MetricType.METERED)
+            .build());
+
+    supportedSessionMetrics.put(
+        THROTTLING_QUEUE_SIZE,
+        Metadata.builder()
+            .withName(THROTTLING_QUEUE_SIZE.getPath())
+            .withDescription("The size of the throttling queue.")
+            .withType(MetricType.GAUGE)
+            .build());
+
+    supportedSessionMetrics.put(
+        CQL_CLIENT_TIMEOUTS,
+        Metadata.builder()
+            .withName(CQL_CLIENT_TIMEOUTS.getPath())
+            .withDescription("The number of CQL requests that timed out.")
+            .withType(MetricType.COUNTER)
+            .build());
+
+    supportedSessionMetrics.put(
+        THROTTLING_ERRORS,
+        Metadata.builder()
+            .withName(THROTTLING_ERRORS.getPath())
+            .withDescription(
+                "The number of times a request was rejected with a RequestThrottlingException.")
+            .withType(MetricType.COUNTER)
+            .build());
+
+    supportedSessionMetrics.put(
+        CQL_PREPARED_CACHE_SIZE,
+        Metadata.builder()
+            .withName(CQL_PREPARED_CACHE_SIZE.getPath())
+            .withDescription("The size of the driver-side cache of CQL prepared statements.")
+            .withType(MetricType.GAUGE)
+            .build());
```

Comment:
There's _a lot_ of code duplication above.  Can reduce this by iterating over DefaultSessionMeric.values()